Fix a unit test failure in non English languages

masap opened the pull request:
#### What type of PR is this?
/kind failing-test

#### What this PR does / why we need it:
Fix a unit test failure in non English languages. For example, this is a result in Japanese language.
```
$ make test WHAT=./staging/src/k8s.io/kubectl/pkg/cmd/diff
[0402 07:24:05] Running tests without code coverage
FAIL: TestDiffProgram (0.00s)
    diff_test.go:73: stdout = "ファイル /dev/zero と /dev/zero は同一です\n", expected = Files /dev/zero and /dev/zero are identical
        "
FAIL
FAIL	k8s.io/kubernetes/staging/src/k8s.io/kubectl/pkg/cmd/diff	0.045s
FAIL
make: *** [Makefile:184: test] エラー 1
```

#### Which issue(s) this PR fixes:
NONE

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs

```

wzshiming commented on staging/src/k8s.io/kubectl/pkg/cmd/diff/diff_test.go:
```diff
@@ -54,6 +54,7 @@ func (f *FakeObject) Live() runtime.Object {
 func TestDiffProgram(t *testing.T) {
 	externalDiffCommands := [3]string{"diff", "diff -ruN", "diff --report-identical-files"}
 
+	os.Setenv("LANG", "C")
```
```suggestion
	oriLang := os.Getenv("LANG")
	os.Setenv("LANG", "C")
	defer os.Setenv("LANG", oriLang)
```

wzshiming replied:
Suggested restore the original environment variables after the test is completed, so as not to affect other cases

masap replied:
OK, fixed.

masap committed:
Fix a unit test failure in non English languages

This is a result in Japanese language.

$ make test WHAT=./staging/src/k8s.io/kubectl/pkg/cmd/diff
[0402 07:24:05] Running tests without code coverage
FAIL: TestDiffProgram (0.00s)
    diff_test.go:73: stdout = "ファイル /dev/zero と /dev/zero は同一です\n", expected = Files /dev/zero and /dev/zero are identical
        "
FAIL
FAIL	k8s.io/kubernetes/staging/src/k8s.io/kubectl/pkg/cmd/diff	0.045s
FAIL
make: *** [Makefile:184: test] エラー 1

masap commented:
/test pull-kubernetes-e2e-kind

masap commented:
/test pull-kubernetes-e2e-kind

masap commented:
/test pull-kubernetes-e2e-kind

dougsland reviewed (commented):
/lgtm
/cc @rikatz

masap commented:
> Question: Are there other tests impacted by this same issue? It's worth to take a look if the localization impacts in other places :D

Other unit tests in `make test` and integration tests in `make test-integration` works fine in Japanese environment (`LANG=ja_JP.UTF-8`).

rikatz reviewed (approved):
/lgtm

rikatz commented:
Thanks!

masap commented:
/test pull-kubernetes-integration

masap commented:
/test pull-kubernetes-e2e-gce-ubuntu-containerd

masap commented:
/test pull-kubernetes-integration
/test pull-kubernetes-e2e-kind

masap commented:
It seems that the test pull-kubernetes-e2e-gce-ubuntu-containerd is broken.

#100978

/hold

masap commented:
/unhold

masap commented:
/test pull-kubernetes-e2e-gce-ubuntu-containerd
